app/components/answer-step.jsx

```diff
@@ -0,0 +1,152 @@
+'use strict'
+import React, { forwardRef, useState, useEffect } from 'react'
+import StepIntro from './step-intro'
+import WhyInput from './why-input'
+import cx from 'classnames'
+import { createUseStyles } from 'react-jss'
+import _ from 'lodash'
+
+const AnswerStep = forwardRef((props, ref) => {
+  const {
+    className = '',
+    intro = '',
+    question = {},
+    whyQuestion = '',
+    shared = {
+      startingPoint: {},
+      whyMosts: [],
+    },
+    onDone = () => {},
+    ...otherProps
+  } = props
+  const classes = useStylesFromThemeFunction()
+
+  // user input to answer important issue
+  const [startingPoint, setStartingPoint] = useState(shared.startingPoint)
+
+  // user input to explain why issue is important
+  const [whyMosts, setWhyMosts] = useState(shared.whyMosts)
+
+  useEffect(() => {
+    if (areStartingComplete() && areWhyAnswersComplete()) {
+      onDone({ valid: true, value: { startingPoint: startingPoint, whyMosts: whyMosts } })
+    } else {
+      onDone({ valid: false, value: { startingPoint: startingPoint, whyMosts: whyMosts } })
+    }
+  }, [whyMosts])
+
+  const updateQuestionResponse = ({ valid, value }) => {
+    const updatedAnswers = {
+      answerSubject: value.subject,
+      answerDescription: value.description,
+      _id: startingPoint._id,
+    }
+    setStartingPoint(updatedAnswers)
+  }
+
+  const updateWhyResponse = ({ valid, value }) => {
+    const updatedAnswers = whyMosts.map(answer => {
+      // find which why response to update
+      if (answer._id === value.parentId) {
+        answer.answerSubject = value.subject
+        answer.answerDescription = value.description
+        answer.valid = valid
+      }
+      return answer
+    })
+    setWhyMosts(updatedAnswers)
+  }
+
+  // evaluates if both subject and description fields are filled out for starting point
+  const areStartingComplete = () => {
+    return startingPoint.answerSubject && startingPoint.answerDescription
+  }
+
+  // evaluates if both subject and description fields are filled out for whyMosts
+  const areWhyAnswersComplete = () => {
+    return whyMosts.every(answer => answer.valid)
+  }
+
+  return (
+    <div className={cx(classes.wrapper, className)} {...otherProps}>
+      <StepIntro subject="Answer" description="Please provide a title and short description of your answer." />
+      <div className={classes.answersContainer}>
+        {question ? (
+          <div key={startingPoint._id}>
+            <WhyInput
+              point={{ subject: '', description: question, _id: startingPoint._id }}
+              defaultValue={{ subject: '', description: '' }}
+              onDone={updateQuestionResponse}
+            />
+          </div>
+        ) : (
+          <div className={classes.noPointsContainer}>
+            <hr className={classes.pointsHr}></hr>
+            There are no questions to respond to.
+          </div>
+        )}
+        {whyMosts.length ? (
```

Comment:
We only want to ask for one whyMost.  "Why is this answer the most important" Don't display the other ones here.

Reply:
I'm confused on this. On one of the story components you want me to have one with multiple points, so that the user can explain multiple times of a problem they want fixed. Do you want me to just delete that story component so it only asks for one reason? If so, should the whyMost prop be an object now instead of an arrray?

Reply:
It's not clear which specific story component you mean, but in the answer step, which is the first step, we want the user to answer the question - and say why they think that answer is the most important for the country to consider.  1 answer, 1 why.

In another step, we are going to show them other whys, and that's why whyMosts in an array.  But there user is only going to create one why for this question, all the other why's will come for other users. And they aren't fetched or displayed until a later step.

And one of the test cases for this component is that there are multiple entries in whyMosts, but this component has to pick out the one that they user authored and only show that one.  This if for the case when the user returns to this step.

Does that help?

Reply:
I see. So the test cases for multiple entries is supposed to have both fields prefilled by responses the user previously put?

Reply:
yes

Reply:
How exactly do I do that? I can just write a test that picks out the appropiate element from the whyMost array. Do I assume that there is going to be a later step that someone will implement a sophisticated algorithm that gets the user's information and pick the respective element in the whyMost array??